src/app/public/modules/auth-http-client/auth-interceptor.ts

```diff
@@ -88,14 +92,24 @@ export class SkyAuthInterceptor implements HttpInterceptor {
       return Observable
         .fromPromise(this.tokenProvider.getContextToken(tokenContextArgs))
         .switchMap((token) => {
-          let authRequest = request.clone({
-            setHeaders: {
-              Authorization: `Bearer ${token}`
-            },
-            url: this.config.runtime.params.getUrl(request.url)
-          });
-
-          return next.handle(authRequest);
+          const decodedToken = this.tokenProvider.decodeToken(token);
+          const zone = decodedToken['1bb.zone'] || '';
+          return Observable
+            .fromPromise(
+              BBAuth.getUrl(request.url,
+              {
+                zone: zone.replace('-', '')
```

Comment:
This seems a little odd to me; how does the caller of `BBAuth.getUrl()` necessarily know that the zone in the JWT is not what should be passed to the method? If this logic is required, shouldn't `BBAuth.getUrl()` have the logic to do any required replacements?